src/test/java/com/hpe/adm/octane/ideplugins/integrationtests/services/RequirementsITCase.java

```diff
@@ -13,41 +13,87 @@
+
+    @Inject
+    private EntityUtils entityUtils;
+
+    private List<EntityModel> entityModelList;
+
+    @Before
+    public void setUp() {
+        ServiceModule serviceModule = TestServiceModule.getServiceModule();
+        Injector injector = Guice.createInjector(serviceModule);
+        injector.injectMembers(this);
+        entityModelList = new ArrayList<>();
+    }
 
     @Test
     public void testCreateRequirement() {
         if (OctaneVersion.compare(versionService.getOctaneVersion(), OctaneVersion.Operation.LOWER_EQ, OctaneVersion.EVERTON_P3)) {
-            EntityModel requirementFolder = createRequirementFolder("folder " + UUID.randomUUID());
-            EntityModel requirement = createRequirement("requirement " + UUID.randomUUID().toString(), requirementFolder);
-            EntityModel createdRequirement = findRequirementById(Long.parseLong(requirement.getValue("id").getValue().toString()));
+            EntityModel requirementFolder = requirementUtils.createRequirementFolder("folder " + UUID.randomUUID());
+            EntityModel requirement = requirementUtils.createRequirement("requirement " + UUID.randomUUID().toString(), requirementFolder);
+            EntityModel createdRequirement = requirementUtils.findRequirementById(Long.parseLong(requirement.getValue("id").getValue().toString()));
+            entityModelList.add(requirementFolder);
+            entityModelList.add(requirement);
             assert (Long.parseLong(requirement.getValue("id").getValue().toString()) == Long.parseLong(createdRequirement.getValue("id").getValue().toString()));
         }
     }
 
     @Test
     public void testSearchRequirement() {
         if (OctaneVersion.compare(versionService.getOctaneVersion(), OctaneVersion.Operation.LOWER_EQ, OctaneVersion.EVERTON_P3)) {
-            EntityModel requirementFolder = createRequirementFolder("folder" + UUID.randomUUID());
-            EntityModel entityModel = createRequirement("requirement " + UUID.randomUUID().toString(), requirementFolder);
+            EntityModel requirementFolder = requirementUtils.createRequirementFolder("folder" + UUID.randomUUID());
+            EntityModel requirement = requirementUtils.createRequirement("requirement " + UUID.randomUUID().toString(), requirementFolder);
+            entityModelList.add(requirementFolder);
+            entityModelList.add(requirement);
             String descriptionText = UUID.randomUUID().toString();
-            setDescription(entityModel, descriptionText);
-            EntityModel createdRequirement = findRequirementById(Long.parseLong(entityModel.getValue("id").getValue().toString()));
+            entityUtils.setDescription(requirement, descriptionText);
+            EntityModel createdRequirement = requirementUtils.findRequirementById(Long.parseLong(requirement.getValue("id").getValue().toString()));
             try {
                 Thread.sleep(40000);//--wait until the elastic search is updated with the entities
```

Comment:
IMO, since we're doing this, we should maybe not set a fixed timeout for this, instead we can do a repeated check with a max fail count, with a set time between tries. (5 sec for example, for a maximum of 15 tries)